presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -0,0 +1,404 @@
+            assertThat(statisticsAfterCreate.getNumFiles().getAsLong()).isEqualTo(50);
+        }
+        finally {
+            onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        }
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, HIVE_TABLE_STATISTICS})
+    public void testInsertBucketedPartitioned()
+    {
+        String tableName = "test_basic_statistics_bucketed_partitioned_insert_presto";
+
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        onPresto().executeQuery(format("" +
+                "CREATE TABLE %s " +
+                "WITH ( " +
+                "   partitioned_by = ARRAY['n_regionkey'], " +
+                "   bucketed_by = ARRAY['n_nationkey'], " +
+                "   bucket_count = 10 " +
+                ") " +
+                "AS " +
+                "SELECT n_nationkey, n_name, n_comment, n_regionkey " +
+                "FROM nation " +
+                "WHERE n_regionkey = 1", tableName));
+
+        try {
+            BasicStatistics tableStatistics = getBasicStatisticsForTable(onHive(), tableName);
+            assertThatStatisticsAreNotPresent(tableStatistics);
+
+            BasicStatistics firstPartitionStatistics = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+            assertThatStatisticsAreNonZero(firstPartitionStatistics);
+            assertThat(firstPartitionStatistics.getNumRows().getAsLong()).isEqualTo(5);
+            assertThat(firstPartitionStatistics.getNumFiles().getAsLong()).isEqualTo(10);
+
+            onPresto().executeQuery(format("" +
+                    "INSERT INTO %s (n_nationkey, n_regionkey, n_name, n_comment) " +
+                    "SELECT n_nationkey, n_regionkey, n_name, n_comment " +
+                    "FROM nation " +
+                    "WHERE n_regionkey = 2", tableName));
+
+            BasicStatistics secondPartitionStatistics = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=2");
+            assertThat(secondPartitionStatistics.getNumRows().getAsLong()).isEqualTo(5);
+            assertThat(secondPartitionStatistics.getNumFiles().getAsLong()).isEqualTo(10);
+
+            // Insert into existing bucketed partition is not supported
+            assertThatThrownBy(() -> insertNationData(onPresto(), tableName))
+                    .hasMessageContaining("Cannot insert into existing partition of bucketed Hive table");
+
+            BasicStatistics secondPartitionUpdatedStatistics = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=2");
+            assertThat(secondPartitionUpdatedStatistics.getNumRows().getAsLong()).isEqualTo(5);
+            assertThat(secondPartitionUpdatedStatistics.getNumFiles().getAsLong()).isEqualTo(10);
+        }
+        finally {
+            onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        }
+    }
+
+    private static void insertNationData(QueryExecutor executor, String tableName)
```

Comment:
I think it makes sense to make `insertNationData` ,`getBasicStatisticsForTable`, and `getBasicStatisticsForPartition` not static so that they can call `onHive` and `onPresto` directly.

That will be easier to read.

But I don't feel strongly. Up to you.